### PR TITLE
Closes #2399: Partial Preloading doesn't create a mobile file

### LINKED ISSUE #2399: Partial Preloading doesn't create a mobile file in 3.5
When updating a post, partial preloading will update the cached file.

**Expected behavior**

In 3.5, I'd expect partial preloading would create cached files for both desktop and mobile when the following options are enabled:
1. **Preload**
2. **Separate cache for mobile devices**

**To Reproduce**
1. Make sure the previously mentioned options are enabled.
2. Update a post.
3. Check the post's folder in `/cache/wp-rocket/`.
4. Only the desktop version will be there.

**Backlog Grooming**
- [x] Reproduce the problem
- [x] Identify the root cause
- [x] Scope a solution
- [x] Estimate the effort

### PR DESCRIPTION
**Pull request by Screenfeed**
This allows to preload (also) the mobile version of the URLs after:
- clearing a post cache,
- clearing a term cache,
- clearing automatically expired cache files.

Closes #2399.

- [ ] QA tests

### CODE REVIEW

File: tests/Integration/inc/Engine/Preload/PartialPreloadSubscriber/maybeDispatch.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace WP_Rocket\Tests\Integration\inc\Engine\Preload\PartialPreloadSubscriber;
+
+use WP_Rocket\Tests\Fixtures\inc\Engine\Preload\PartialProcess_Wrapper;
+use WPMedia\PHPUnit\Integration\TestCase;
+
+/**
+ * @covers \WP_Rocket\Engine\Preload\PartialPreloadSubscriber::maybeDispatch
```
Comment (Tabrisrp) [inline tests/Integration/inc/Engine/Preload/PartialPreloadSubscriber/maybeDispatch.php]:
```suggestion
 * @covers \WP_Rocket\Engine\Preload\PartialPreloadSubscriber::maybe_dispatch
```

**Comment by Tabrisrp** [comment]
@piotrbak This is ready for QA

Partial preloading shoud now generate the mobile version of the cache when preload and mobile cache are enabled.

**Comment by piotrbak** [comment]
Starting QA

**Comment by hellofromtonya** [comment]
>However, it seems that we're clearing too many files when there are relations parent-child in categories. Moreover, updating the category description clears its cache but doesn't preload it. I'll investigate it asap.

@piotrbak 

- Did you discover what's happening here?
- Does this happen as a result of this PR, i.e. meaning it doesn't happen without this PR?

**Comment by piotrbak** [comment]
@hellofromtonya 

> Did you discover what's happening here?
Not yet, when it comes to the categories cache clearing, I asked for advice in `l2` channel.

> Does this happen as a result of this PR, i.e. meaning it doesn't happen without this PR?
I was able to determine today that it's not related to this specific PR.

**Comment by hellofromtonya** [comment]
>I was able to determine today that it's not related to this specific PR.

@piotrbak Thank you. I'll go ahead and merge this PR. 

If there's an issue related to categories cache clearing, then please open a new GH issue. Thanks.

**Pull request merged**